vividus/src/main/java/org/vividus/runner/DeprecatedPropertiesReplacer.java

```diff
@@ -0,0 +1,118 @@
+import org.vividus.configuration.DeprecatedPropertiesHandler;
+import org.vividus.configuration.Vividus;
+
+public class DeprecatedPropertiesReplacer
+{
+    private static final String PLACEHOLDER_PREFIX = "${";
+    private static final String PLACEHOLDER_SUFFIX = "}";
+    private static final String DEPRECATED_PROPERTIES_CLASSPATH = "classpath*:/properties/**/deprecated.properties";
+
+    private static final Properties DEPRECATED_PROPERTIES = new Properties();
+
+    private final ResourcePatternResolver resourcePatternResolver;
+
+    public DeprecatedPropertiesReplacer(ResourcePatternResolver resourcePatternResolver)
+    {
+        this.resourcePatternResolver = resourcePatternResolver;
+    }
+
+    public static void main(String[] args) throws IOException, ParseException
+    {
+        Vividus.init();
+
+        Options options = new Options();
+        Option directoryOption = new Option("rl", "resourceLocation", true, "Directory to find properties.");
+        options.addOption(directoryOption);
+        CommandLineParser parser = new DefaultParser();
+        CommandLine commandLine = parser.parse(options, args);
+        String testResourcesLocation = commandLine.getOptionValue(directoryOption.getOpt());
+
+        DeprecatedPropertiesReplacer deprecatedPropertiesReplacer = new DeprecatedPropertiesReplacer(
+                BeanFactory.getResourcePatternResolver());
+        deprecatedPropertiesReplacer.replaceDeprecatedProperties(testResourcesLocation);
+    }
+
+    private void replaceDeprecatedProperties(String testResourcesLocation) throws IOException
+    {
+        loadDeprecatedProperties();
+        Resource[] allPropertiesAsResources = resourcePatternResolver
+                .getResources(Path.of(testResourcesLocation).toUri() + "/**/*.properties");
+
+        for (Resource resource : allPropertiesAsResources)
+        {
+            replaceDeprecatedPropertiesInResource(resource);
+        }
+    }
+
+    private void replaceDeprecatedPropertiesInResource(Resource resource) throws IOException
+    {
+        String propertiesAsString = resource.getContentAsString(StandardCharsets.UTF_8);
+        Properties currentProperties = new Properties();
+        currentProperties.load(new StringReader(propertiesAsString));
+        Properties actualProperties = (Properties) currentProperties.clone();
+
+        DeprecatedPropertiesHandler deprecatedPropertiesHandler = new DeprecatedPropertiesHandler(DEPRECATED_PROPERTIES,
+                PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+        deprecatedPropertiesHandler.replaceDeprecated(actualProperties);
+
+        for (Map.Entry<Object, Object> entry : currentProperties.entrySet())
+        {
+            String oldKey = (String) entry.getKey();
+            String actualKey = deprecatedPropertiesHandler.getActualPropertyKey(oldKey);
+
+            if (actualKey != null)
+            {
+                propertiesAsString = propertiesAsString.replace(oldKey, actualKey);
```

Comment:
can we do replace all? I think some properties might be commented just for testing purposes